core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -125,41 +124,12 @@ public class GenericContainer<SELF extends GenericContainer<SELF>>
     /*
      * Default settings
      */
-    @NonNull
-    private LinkedHashSet<Integer> exposedPorts = new LinkedHashSet<>();
-
-    @NonNull
-    private List<String> portBindings = new ArrayList<>();
-
     @NonNull
     private List<String> extraHosts = new ArrayList<>();
 
-    @NonNull
-    private String networkMode;
-
-    @Nullable
-    private Network network;
-
-    @NonNull
-    private List<String> networkAliases = new ArrayList<>(Arrays.asList("tc-" + Base58.randomString(8)));
-
     @NonNull
     private RemoteDockerImage image;
 
-    @NonNull
-    private Map<String, String> env = new HashMap<>();
-
-    @NonNull
-    private Map<String, String> labels = new HashMap<>();
-
-    @NonNull
-    private String[] commandParts = new String[0];
-
-    @NonNull
-    private List<Bind> binds = new ArrayList<>();
-
-    private boolean privilegedMode;
-
     @NonNull
     private List<VolumesFrom> volumesFroms = new ArrayList<>();
```

Comment:
This should be probably moved into `ContainerDef`. But I am unsure whether this feature is really used by users in practice.

Reply:
Sounds like a very rarely used feature that should be considered through `CreateContainerModifier` instead 👀

Reply:
Will we move it in a follow-up PR to avoid breaking changes?